app/decorators/service_provider_session_decorator.rb

```diff
@@ -48,4 +52,18 @@ def sp_return_url
   private
 
   attr_reader :sp, :view_context
+
+  def sp_logo_file_exist?
```

Comment:
I would vote for broken image. It's in your face and clear that something is wrong. That code scares me.